tests/test_lando.py

```diff
@@ -0,0 +1,122 @@
+        "group": MOCK_GROUP,
+        "id": MOCK_WARNINGS_ID,
+    }
+]
+
+
+def mocked_get_warnings(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, json_data, status_code):
+            self.json_data = json_data
+            self.status_code = status_code
+
+        def json(self):
+            return self.json_data
+
+    # Test URL passed to requests.get
+    assert args[0] == MOCK_LANDO_API_URL + "/diff_warnings"
+
+    # Test the `params` passed to requests.get
+    assert kwargs["params"] == {
+        "revision_id": MOCK_REV_ID,
+        "diff_id": MOCK_DIFF_ID,
+        "group": MOCK_GROUP,
+    }
+
+    # Test the `headers`
+    assert kwargs["headers"] == {"X-Phabricator-API-Key": MOCK_LANDO_TOKEN}
+
+    return MockResponse(MOCK_WARNINGS_DATA, 200)
+
+
+def mocked_add_warning(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, status_code):
+            self.status_code = status_code
+
+    # Test URL passed to requests.post
+    assert args[0] == MOCK_LANDO_API_URL + "/diff_warnings"
+
+    # Test the `params` passed to requests.get
+    assert kwargs["json"] == {
+        "revision_id": MOCK_REV_ID,
+        "diff_id": MOCK_DIFF_ID,
+        "group": MOCK_GROUP,
+        "data": {"message": MOCK_WARNING_MESSAGE},
+    }
+
+    # Test the `headers`
+    assert kwargs["headers"] == {"X-Phabricator-API-Key": MOCK_LANDO_TOKEN}
+
+    return MockResponse(201)
+
+
+def mocked_del_warnings(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, status_code):
+            self.status_code = status_code
+
+    # Test URL passed to requests.del
+    assert args[0] == MOCK_LANDO_API_URL + "/diff_warnings/" + str(MOCK_WARNINGS_ID)
+
+    # Test the `headers`
+    assert kwargs["headers"] == {"X-Phabricator-API-Key": MOCK_LANDO_TOKEN}
+
+    return MockResponse(200)
+
+
+class Test_LandoWarnings(unittest.TestCase):
+    @mock.patch("requests.get", side_effect=mocked_get_warnings)
+    def test_get_warnings(self, mock_get):
+
+        mock_lando = LandoWarnings(MOCK_LANDO_API_URL, MOCK_LANDO_TOKEN)
+
+        self.assertEqual(
+            mock_lando.get_warnings(MOCK_REV_ID, MOCK_DIFF_ID), MOCK_WARNINGS_DATA
+        )
+
+    @mock.patch("requests.post", side_effect=mocked_add_warning)
+    def test_add_warnings(self, mock_post):
+
+        mock_lando = LandoWarnings(MOCK_LANDO_API_URL, MOCK_LANDO_TOKEN)
+
+        self.assertEqual(
+            mock_lando.add_warning(MOCK_WARNING_MESSAGE, MOCK_REV_ID, MOCK_DIFF_ID),
+            LandoWarnings.ReturnCode.OK,
+        )
+
+    @mock.patch("requests.delete", side_effect=mocked_del_warnings)
+    @mock.patch("requests.get", side_effect=mocked_get_warnings)
+    def test_del_all_warnings(self, mock_del, moc_get):
+
+        mock_lando = LandoWarnings(MOCK_LANDO_API_URL, MOCK_LANDO_TOKEN)
+
+        self.assertEqual(
+            mock_lando.del_all_warnings(MOCK_REV_ID, MOCK_DIFF_ID),
+            LandoWarnings.ReturnCode.OK,
+        )
+
+
```

Comment:
We should also test some error conditions

Reply:
Do you have something particular in mind?